### ISSUE TITLE
[Due for payment 2026-02-17] [$3000] Chat - Most of the workspace modifying settings messages are not translated in #Admins

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#52984]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.66-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

**Issue found when executing PR** https://github.com/Expensify/App/pull/52159
## Action Performed:
Setup: Login on ND with a user that has a workspace where some settings have been changed (for instance, currency, reporting frequency, approval mode, approver)
1. At the bottom of the screen, go to "Settings"
2. Click on "Preferences"
3. Click on "Language" and select "Spanish"
4. Go to the "#Admins" room

## Expected Result:
All workspace modifying settings messages are translated.
For instance:
SP: actualizó la moneda por defecto de ARS - AR$ a USD - $
EN: updated the default currency from ARS - AR$ to USD - $

## Actual Result:
Most of the workspace modifying settings messages are not translated

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

![Bug6673572_1732310423428!messages_not_translated](https://github.com/user-attachments/assets/10ed9fd8-d6fc-48c7-a3de-17f5b3100060)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021861045305847069497</li>
        <li>Upwork Job ID: 1861045305847069497</li>
        <li>Last Price Increase: 2026-02-06</li>
<li>Automatic offers: </li>
<ul>
<li>Krishna2323 | Contributor | 105059490</li>
<li>abzokhattab | Contributor | 109317981</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @garrettmknight / jliexpensify</details>

**Comment by Shahidullah-Muffakir** [comment]
We are getting these message from the api response.

**Comment by jliexpensify** [comment]
Hi @rushatgabhane @techievivek - assigning you for review as you worked on https://github.com/Expensify/App/pull/52159.

**Comment by techievivek** [comment]
@Krishna2323 would you be interested in working on this one as well? Can you take a look at the GH?

**Comment by techievivek** [comment]
Sounds good to me.

**Comment by techievivek** [comment]
Not overdue, PR is being worked on.

**Comment by jliexpensify** [comment]
Done, thanks for setting expectations @Krishna2323 - @techievivek can change it back if it's urgent

**Comment by stephanieelliott** [comment]
Hey @Krishna2323, I have an issue where the default workspace name is not being translated when Spanish language is selected (i.e. `My Group Workspace` should be `Mi Espacio de Trabajo en Grupo`). I know you are working on a handful of WS translations here -- will the workspace name be addressed in your PR, or should we handle that separately? https://github.com/Expensify/App/issues/53599

**Comment by jliexpensify** [comment]
I'm OOO until the 29th, feel free to reassign if payment is needed before then - otherwise will catch up when I get back.

**Comment by jliexpensify** [comment]
@Krishna2323 you're still working on this right?

**Comment by jliexpensify** [comment]
Re: pricing, I think that's reasonable.

Re: splitting, I think let's defer to @techievivek (I'm not an Engineer so I'm unsure how to best manage this)

**Comment by techievivek** [comment]
Yeah, agree with both the suggestions. Let's push 2 PRs here.

**Comment by jliexpensify** [comment]
I've increased the bounty to $1000 cc @techievivek @rushatgabhane

**Comment by rushatgabhane** [comment]
@Krishna2323 could you please link the PR? I can't find it

**Comment by techievivek** [comment]
How is PR going here? @Krishna2323, do you need help with anything, please let me know if you do.

**Comment by jliexpensify** [comment]
Thanks for the heads up! I'll remove payment date for now.

**Comment by jliexpensify** [comment]
Not overdue. Making this a `Weekly` as we wait on PR #2

**Comment by jliexpensify** [comment]
How are we going here @Krishna2323?

**Comment by techievivek** [comment]
@Krishna2323 Let me know if you need help with anything, I can check the backend and help you with the reproduction steps.

**Comment by techievivek** [comment]
Yeah, I agree—it’s better to keep the PRs smaller. We can definitely create a third or even a fourth one if needed. There are a lot of settings and sub-settings options available for this flow, so the important thing is to keep making progress. The exact number of PRs doesn’t really matter.

**Comment by rushatgabhane** [comment]
@Krishna2323 how's it looking? 😄

**Comment by jliexpensify** [comment]
Not overdue

**Comment by jliexpensify** [comment]
Damn, this took ages - nice one @Krishna2323!

**Comment by jliexpensify** [comment]
@Krishna2323 invited you to the new job: https://www.upwork.com/jobs/~021934450119164795023

**Comment by rushatgabhane** [comment]
> do we need a checklist? And are we also paying you $1000?

yes and yes because i reviewed all the PRs

**Comment by jliexpensify** [comment]
Bump @Krishna2323 please accept the new job in Upworks!

**Comment by jliexpensify** [comment]
@rushatgabhane bump on checklist please.

**Comment by Krishna2323** [comment]
@jliexpensify, apologies for the confusion—I was OOO last week. Could you please check [this comment](https://github.com/Expensify/App/issues/52984#issuecomment-2785408809)? We still have 30 translations (7 new system message has been added) remaining.

I’ll aim to complete them within the next two weeks, though I may need assistance from @techievivek for a few system messages that are tricky to reproduce.

<details>
## For tracking:

### Completed:
ADD_APPROVER_RULE: 'POLICYCHANGELOG_ADD_APPROVER_RULE — ✅',
ADD_CATEGORY: 'POLICYCHANGELOG_ADD_CATEGORY — ✅',
ADD_CUSTOM_UNIT_RATE: 'POLICYCHANGELOG_ADD_CUSTOM_UNIT_RATE — ✅',
ADD_EMPLOYEE: 'POLICYCHANGELOG_ADD_EMPLOYEE — ✅',
ADD_INTEGRATION: 'POLICYCHANGELOG_ADD_INTEGRATION — ✅',
ADD_REPORT_FIELD: 'POLICYCHANGELOG_ADD_REPORT_FIELD — ✅',
ADD_TAG: 'POLICYCHANGELOG_ADD_TAG — ✅',
DELETE_APPROVER_RULE: 'POLICYCHANGELOG_DELETE_APPROVER_RULE — ✅',
DELETE_CATEGORY: 'POLICYCHANGELOG_DELETE_CATEGORY — ✅',
DELETE_CUSTOM_UNIT_RATE: 'POLICYCHANGELOG_DELETE_CUSTOM_UNIT_RATE — ✅',
DELETE_EMPLOYEE: 'POLICYCHANGELOG_DELETE_EMPLOYEE — ✅',
DELETE_INTEGRATION: 'POLICYCHANGELOG_DELETE_INTEGRATION — ✅',
DELETE_REPORT_FIELD: 'POLICYCHANGELOG_DELETE_REPORT_FIELD — ✅',
DELETE_TAG: 'POLICYCHANGELOG_DELETE_TAG — ✅',
SET_CATEGORY_NAME: 'POLICYCHANGELOG_SET_CATEGORY_NAME — ✅',
UPDATE_APPROVER_RULE: 'POLICYCHANGELOG_UPDATE_APPROVER_RULE — ✅',
UPDATE_AUDIT_RATE: 'POLICYCHANGELOG_UPDATE_AUDIT_RATE — ✅',
UPDATE_AUTO_REPORTING_FREQUENCY: 'POLICYCHANGELOG_UPDATE_AUTOREPORTING_FREQUENCY — ✅',
UPDATE_CATEGORY: 'POLICYCHANGELOG_UPDATE_CATEGORY — ✅',
UPDATE_CURRENCY: 'POLICYCHANGELOG_UPDATE_CURRENCY — ✅',
UPDATE_CUSTOM_UNIT: 'POLICYCHANGELOG_UPDATE_CUSTOM_UNIT — ✅',
UPDATE_CUSTOM_UNIT_RATE: 'POLICYCHANGELOG_UPDATE_CUSTOM_UNIT_RATE — ✅',
UPDATE_DEFAULT_BILLABLE: 'POLICYCHANGELOG_UPDATE_DEFAULT_BILLABLE — ✅',
UPDATE_DEFAULT_TITLE: 'POLICYCHANGELOG_UPDATE_DEFAULT_TITLE — ✅',
UPDATE_DEFAULT_TITLE_ENFORCED: 'POLICYCHANGELOG_UPDATE_DEFAULT_TITLE_ENFORCED — ✅',
UPDATE_EMPLOYEE: 'POLICYCHANGELOG_UPDATE_EMPLOYEE — ✅',
UPDATE_FIELD: 'POLICYCHANGELOG_UPDATE_FIELD — APPROVAL_MODE ✅🟡 PREVENT_SELF_APPROVAL ✅ MAX_EXPENSE_AGE ✅',
UPDATE_MANUAL_APPROVAL_THRESHOLD: 'POLICYCHANGELOG_UPDATE_MANUAL_APPROVAL_THRESHOLD — ✅',
UPDATE_MAX_EXPENSE_AMOUNT: 'POLICYCHANGELOG_UPDATE_MAX_EXPENSE_AMOUNT — ✅',
UPDATE_MAX_EXPENSE_AMOUNT_NO_RECEIPT: 'POLICYCHANGELOG_UPDATE_MAX_EXPENSE_AMOUNT_NO_RECEIPT — ✅',
UPDATE_NAME: 'POLICYCHANGELOG_UPDATE_NAME — ✅',
UPDATE_DESCRIPTION: 'POLICYCHANGELOG_UPDATE_DESCRIPTION — ✅',
UPDATE_REPORT_FIELD: 'POLICYCHANGELOG_UPDATE_REPORT_FIELD — ✅',
UPDATE_TAG: 'POLICYCHANGELOG_UPDATE_TAG — ✅🟡',
UPDATE_TAG_ENABLED: 'POLICYCHANGELOG_UPDATE_TAG_ENABLED — ✅',
UPDATE_TAG_LIST_NAME: 'POLICYCHANGELOG_UPDATE_TAG_LIST_NAME — ✅',
UPDATE_TAG_NAME: 'POLICYCHANGELOG_UPDATE_TAG_NAME — ✅',
LEAVE_POLICY: 'POLICYCHANGELOG_LEAVE_POLICY — ✅',
CORPORATE_UPGRADE: 'POLICYCHANGELOG_CORPORATE_UPGRADE — ✅',
TEAM_DOWNGRADE: 'POLICYCHANGELOG_TEAM_DOWNGRADE — ✅',
DELETE_MULTIPLE_TAGS: 'POLICYCHANGELOG_DELETE_MULTIPLE_TAGS — ✅',

### Remaining: 
ADD_BUDGET: 'POLICYCHANGELOG_ADD_BUDGET — ',
ADD_CUSTOM_UNIT: 'POLICYCHANGELOG_ADD_CUSTOM_UNIT — ',
DELETE_ALL_TAGS: 'POLICYCHANGELOG_DELETE_ALL_TAGS — ',
DELETE_BUDGET: 'POLICYCHANGELOG_DELETE_BUDGET — ',
DELETE_CUSTOM_UNIT: 'POLICYCHANGELOG_DELETE_CUSTOM_UNIT — ',
DELETE_CUSTOM_UNIT_SUB_RATE: 'POLICYCHANGELOG_DELETE_CUSTOM_UNIT_SUB_RATE — ',
IMPORT_CUSTOM_UNIT_RATES: 'POLICYCHANGELOG_IMPORT_CUSTOM_UNIT_RATES — ',
IMPORT_TAGS: 'POLICYCHANGELOG_IMPORT_TAGS — ',
INDIVIDUAL_BUDGET_NOTIFICATION: 'POLICYCHANGELOG_INDIVIDUAL_BUDGET_NOTIFICATION — ',
INVITE_TO_ROOM: 'POLICYCHANGELOG_INVITETOROOM — ',
REMOVE_FROM_ROOM: 'POLICYCHANGELOG_REMOVEFROMROOM — ',
LEAVE_ROOM: 'POLICYCHANGELOG_LEAVEROOM — ',
REPLACE_CATEGORIES: 'POLICYCHANGELOG_REPLACE_CATEGORIES — ',
SET_AUTO_REIMBURSEMENT: 'POLICYCHANGELOG_SET_AUTOREIMBURSEMENT — ',
SET_AUTO_JOIN: 'POLICYCHANGELOG_SET_AUTO_JOIN — ',
SHARED_BUDGET_NOTIFICATION: 'POLICYCHANGELOG_SHARED_BUDGET_NOTIFICATION — ',
UPDATE_ACH_ACCOUNT: 'POLICYCHANGELOG_UPDATE_ACH_ACCOUNT — ',
UPDATE_AUTO_HARVESTING: 'POLICYCHANGELOG_UPDATE_AUTOHARVESTING — ',
UPDATE_AUTO_REIMBURSEMENT: 'POLICYCHANGELOG_UPDATE_AUTOREIMBURSEMENT — ',
UPDATE_BUDGET: 'POLICYCHANGELOG_UPDATE_BUDGET — ',
UPDATE_CATEGORIES: 'POLICYCHANGELOG_UPDATE_CATEGORIES — ',
UPDATE_CUSTOM_UNIT_SUB_RATE: 'POLICYCHANGELOG_UPDATE_CUSTOM_UNIT_SUB_RATE — ',
UPDATE_DEFAULT_REIMBURSABLE: 'POLICYCHANGELOG_UPDATE_DEFAULT_REIMBURSABLE — ',
UPDATE_DISABLED_FIELDS: 'POLICYCHANGELOG_UPDATE_DISABLED_FIELDS — ',
UPDATE_OWNERSHIP: 'POLICYCHANGELOG_UPDATE_OWNERSHIP — ',
UPDATE_PROHIBITED_EXPENSES: 'POLICYCHANGELOG_UPDATE_PROHIBITED_EXPENSES — ',
UPDATE_REIMBURSEMENT_CHOICE: 'POLICYCHANGELOG_UPDATE_REIMBURSEMENT_CHOICE — ',
UPDATE_TAG_LIST_REQUIRED: 'POLICYCHANGELOG_UPDATE_TAG_LIST_REQUIRED — ',
UPDATE_TIME_ENABLED: 'POLICYCHANGELOG_UPDATE_TIME_ENABLED — ',
UPDATE_TIME_RATE: 'POLICYCHANGELOG_UPDATE_TIME_RATE — ',
UPDATE_TAG_LIST: 'POLICYCHANGELOG_UPDATE_TAG_LIST — ',

</details>

**Comment by jliexpensify** [comment]
Oh no stress - the payment summary won't change, so LMK when you're done (as the job may expire). I'll adjust the payment date again.

**Comment by Krishna2323** [comment]
@techievivek @jliexpensify, before moving forward, I want to clarify a few things. Initially, I estimated there were 55–60 translations to add and [requested an increase of the bounty to $1000 for all the translations](https://github.com/Expensify/App/issues/52984#issuecomment-2575327510). However, in the second PR, [I discovered there were actually more translations required (probably 90+), because some changelogs have subtypes](https://github.com/Expensify/App/issues/52984#issuecomment-2785408809). I then requested to split the work into 3 PRs.

At this point, I have already added 58 translations across both PRs, and I honestly feel I have invested significantly more time and effort than I initially estimated. Given that the total number of translations will be around 90, I would like to request an increase of the bounty to $1500. If you feel this is unreasonable, please let me know—I’m willing to complete the work at the original price, but I would feel quite undervalued. For context, we typically pay $125 for just 1–2 translation updates at ATM, so even at $1500 for all 90, this would amount to only about $16.50 per translation.

Sorry for going into detail about the bounty, but I really feel I have put in a great deal of time and effort on this project. There are still about 30 translations left, which may require even more effort than the previous two PRs, as I will likely need additional help to reproduce most of them.

I’m also happy to step aside if you’d prefer to have someone else take over the remaining work. Thanks for understanding. 🙏🏻🙇🏻

**Comment by jliexpensify** [comment]
@Krishna2323 appreciate the insight. I think it's fine to bump to $1500, or even $2000 as you're doing so many - cc @techievivek for thoughts.

**Comment by techievivek** [comment]
Absolutely agree with the above comment. @Krishna2323 has put in a good amount of effort to bring this to the finish line.

**Comment by jliexpensify** [comment]
Adjusted to $2000 for @Krishna2323 . @techievivek is reviewing PR's the same amount of effort? i.e. Should we give 2k to @rushatgabhane as well?

**Comment by garrettmknight** [comment]
$2000 approved for @rushatgabhane

**Comment by garrettmknight** [comment]
@jliexpensify sounds like this isn't ready to be paid out yet.

**Comment by jliexpensify** [comment]
Not overdue!

**Comment by techievivek** [comment]
Still need to go through the comment here https://github.com/Expensify/App/issues/52984#issuecomment-3015992499 and help Krishna with reproduction steps.

**Comment by techievivek** [comment]
Added myself as the issue owner so it gets shown on my K2.

**Comment by techievivek** [comment]
Not overdue, working on it.

**Comment by jliexpensify** [comment]
@techievivek @Krishna2323 how's everytihng going?

**Comment by techievivek** [comment]
😬 I will need to test this on my end then.

**Comment by tgolen** [comment]
Reassigning CME due to OOO

@blimpich I think you'll need to pick up [here](https://github.com/Expensify/App/issues/52984#issuecomment-3380806796)

**Comment by jliexpensify** [comment]
Payment Summary

- C: @Krishna2323 $2000
- C+: @rushatgabhane $1000
- C+: @abzokhattab $500

**Comment by blimpich** [comment]
not overdue

**Comment by jliexpensify** [comment]
Not overdue

**Comment by jamesdeanexpensify** [comment]
@Krishna2323 are you able to go through the doc and accept suggestions if they look good? Feel free to leave the other comments outstanding so we can see what's left and I'll review again.

**Comment by jamesdeanexpensify** [comment]
@Krishna2323 are you OK to accept all suggestions in the Google Doc (or are you OK with me doing it)? I might have some more changes, so I wouldn't make any updates just yet.

**Comment by jamesdeanexpensify** [comment]
OK - I guess my question is if you need to see the changes in the Google Doc before I accept suggestions. If not, then I'll just accept them. Let me know!

**Comment by jamesdeanexpensify** [comment]
@trjExpensify @JmillsExpensify now that my eyes are crossed, do you mind taking one last once-over of [this doc](https://docs.google.com/document/d/1tJzk23znpnWU2_ad21tDCmdA9yhZ2Pih39XmkLADRQo/edit?pli=1&tab=t.0) when you get a chance? thank you!

**Comment by blimpich** [comment]
Looks like all suggestions have been applied to the document. Bump @trjExpensify @JmillsExpensify speak now or forever hold your peace ☮️

**Comment by jamesdeanexpensify** [comment]
Good to go!

**Comment by Krishna2323** [comment]
@jamesdeanexpensify I have updated the doc: https://docs.google.com/document/d/1tJzk23znpnWU2_ad21tDCmdA9yhZ2Pih39XmkLADRQo/edit?pli=1&tab=t.0#heading=h.1nvcgumncccx

**Comment by jamesdeanexpensify** [comment]
ok @Krishna2323 I think we are looking good now in the doc

**Comment by Krishna2323** [comment]
The PR is currently quite messy. I think I may have removed some parts while resolving conflicts, and I’m checking on it now.

We also have a new pattern now where each function should receive the `translate` function, so we don’t have to use `Onyx.connect.` I need to update all the functions accordingly. I’ll provide updates very soon.

**Comment by Krishna2323** [comment]
@blimpich After completing everything, I started getting an error about the max 4k lines per file limit, so I split the code in `ReportActionsUtils.ts`. It took the whole day, and I’m still resolving conflicts.

What do you think is the best approach here? Should I revert all the file-split changes, or should I proceed with resolving the conflicts? I’m fine with either option, but resolving the conflicts seems tougher at this point.

Honestly, after resolving lint issues and conflicts, I’m pretty exhausted and just wanted your guidance. Another option is to merge this PR with the lint issue as is, and then I can raise a separate PR to split the file into two. WDYT?

**Comment by blimpich** [comment]
Hello @Krishna2323! Sorry this has gone on for so long, totally get getting tired. I will look into this today and get back to you about the best path forward. Need to wrap my head around this first. Feel free to take some rest while I work on this and get back to you.

For reference this is the PR right: https://github.com/Expensify/App/pull/65074

**Comment by Krishna2323** [comment]
> In terms of $$$ - is $2000 still ok?

@JmillsExpensify @blimpich

Sorry to bring up compensation again, but after completing the third PR, I wanted to briefly revisit this. I feel that $2000 may be a bit low given the total effort and time involved across all three PRs.

Across **3 PRs** (#57217, #53238, #65074), we added approximately **81 new English translation keys in `en.ts`** for workspace-modifying settings messages in chat. In addition, we introduced around **38 conditional logic paths** (for example, subkey variations in budget messages, category updates, and other workspace actions), bringing the total number of translations handled to **nearly 120**.

The first two PRs went relatively smoothly, but the third one took significantly more time. For most translation keys, I first had to reproduce the corresponding system logs, which was time-consuming in itself. We also had long Slack threads to clarify reproduction steps and confirm translations, and reviewing such a large number of messages added extra complexity.

Additionally, since the last PR was open for several months, each update required resolving non-trivial merge conflicts in frequently updated files, which added considerable overhead.

I completely understand if the current amount needs to stay as is, but I wanted to share this context so the effort involved is fully visible. Thanks for considering, and I really appreciate all the support throughout this process.

**Comment by blimpich** [comment]
I agree with bumping it. This was a beast of a project and took up a lot of time and energy from @Krishna2323. I think 3k is fair. It took a year to complete, and was passed between CMEs and C+. Krishna was consistent and kept working at it 👍

**Comment by blimpich** [comment]
All other payments should stay the same as per [this comment](https://github.com/Expensify/App/issues/52984#issuecomment-3474627729), only Krishna's should be bumped.

**Comment by Krishna2323** [comment]
Thank you, everyone. I really appreciate the support, guidance, and patience throughout this process. It means a lot. ❤️

@jliexpensify, could I be paid through ND? I’ve been eligible since around March 2025.

**Comment by jliexpensify** [comment]
Payment Summary

- C: @Krishna2323 $3000 (to be paid via New Expensify)
- C+: @rushatgabhane $1000 (to be paid via New Expensify)

Is this correct?

**Comment by huangpei2026psq** [comment]
GitHub username: huangpei2026psq
I'm interested in this $3000 bounty.
PayPal: 595360703@qq.com
ETH: 0x34f5BC9ad80B9D9d64566c22496e0Df7AFD4f662
Please let me know the next steps. Thanks!

**Comment by jliexpensify** [comment]
[Final payment summary (!!!)](https://github.com/Expensify/App/issues/52984#issuecomment-3905648643)

**Comment by madman12304** [comment]
Hi  @jliexpensify  @rushatgabhane @Krishna2323 @abzokhattab  ....  is that issue still available i would be interested to work on it

**Comment by jliexpensify** [comment]
Chatted to Rushat via DM - he said he's sorted for payments!

**Comment by jliexpensify** [comment]
Once again, if anyone needs it - [this should be the final payment summary](https://github.com/Expensify/App/issues/52984#issuecomment-3917738622).

**Comment by Krishna2323** [comment]
[The PR](https://github.com/Expensify/App/pull/65074) for this issue contributed to #98202. The new message was wired into chat, LHN and clipboard, but not report name or last-message paths, so the thread header kept the server text. It only became visible after #97921 changed the copy. Fixed in #98340 / #98376.
